### PR TITLE
vagrant: update to 1.6.1 with a lot of fixes

### PR DESCRIPTION
**Pull request by obnoxxx**
--api-advertise-addresses --> --apiserver-advertise-address

https://github.com/kubernetes/kubernetes/pull/42064

Signed-off-by: Michael Adam <obnox@redhat.com>

---
This change is [Reviewable](https://reviewable.io/reviews/gluster/gluster-kubernetes/227)

### CODE REVIEW

**Comment by obnoxxx** [comment]
ok... this is not enough.

Problem is: we get kubeadm from the kubernetes yum repo. and that has apparently been updated to kubernetes 1.6.0. But kubectl and kubelet we download directly from googleapis... of version 1.4.5. and that does not work together with kubeadm 1.6.

Now since I could not find kubeadm 1.4.5 for download, so i've been trying to get a working setup with everything at 1.6.0.. no luck up to now... keeing on trying...

**Comment by obnoxxx** [comment]
additional (failed) attemt(s) ...

**Comment by jarrpa** [comment]
@obnoxxx This latest batch look like reasonable patches. What's the status?

**Comment by obnoxxx** [comment]
@jarrpa (most of) the patches are reasonable. But is does not work:

* kubeadm ist just broken with kubernetes 1.6.0: https://github.com/kubernetes/kubernetes/issues/43815
* Older versions are not available any more.
* Patched versions (that work for me) are not available yet, as it seems...

**Comment by obnoxxx** [comment]
updated to latest kube 1.6.1, which seems to have fixed most of the issues.

the current patchset lets ./up.sh run to completion, but I currently have the situation that only the master node gets ready, and some of the weave pods keep looping in errors (CrashLoopBackOff):

```
[vagrant@master ~]$ kubectl get nodes
NAME      STATUS     AGE       VERSION
master    Ready      17m       v1.6.1
node0     NotReady   10m       v1.6.1
node1     NotReady   10m       v1.6.1
node2     NotReady   10m       v1.6.1
[vagrant@master ~]$ kubectl get pods --all-namespaces
NAMESPACE     NAME                             READY     STATUS             RESTARTS   AGE
kube-system   etcd-master                      1/1       Running            0          15m
kube-system   kube-apiserver-master            1/1       Running            0          15m
kube-system   kube-controller-manager-master   1/1       Running            0          15m
kube-system   kube-dns-3913472980-cpz0z        3/3       Running            0          16m
kube-system   kube-proxy-0163s                 1/1       Running            0          8m
kube-system   kube-proxy-0d8lh                 1/1       Running            0          16m
kube-system   kube-proxy-1ctnf                 1/1       Running            0          8m
kube-system   kube-proxy-vr4p7                 1/1       Running            0          8m
kube-system   kube-scheduler-master            1/1       Running            0          15m
kube-system   weave-net-j07lq                  1/2       CrashLoopBackOff   5          8m
kube-system   weave-net-msgg7                  1/2       CrashLoopBackOff   5          8m
kube-system   weave-net-ql98b                  2/2       Running            0          16m
kube-system   weave-net-zq3hd                  1/2       CrashLoopBackOff   5          8m
[vagrant@master ~]$ 

```

**Comment by obnoxxx** [comment]
It seems that the port of the api server (6443) is not propagated to the kubernetes service:

api server listening on 6443 only:
```
 sudo netstat -nltp | grep 443
tcp6       0      0 :::6443                 :::*                    LISTEN      15556/kube-apiserve 
```

service claims 443:
```
$ kubectl get svc
NAME         CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
kubernetes   10.96.0.1    <none>        443/TCP   43m
```

causing weave to fail to connect to apiserver:
```
$ kubectl -n kube-system logs weave-net-5xgmz  weave                                                                                                                                               
2017/04/04 09:55:15 error contacting APIServer: Get https://10.96.0.1:443/api/v1/nodes: dial tcp 10.96.0.1:443: i/o timeout; trying with fallback: http://localhost:8080
2017/04/04 09:55:15 Could not get peers: Get http://localhost:8080/api/v1/nodes: dial tcp 127.0.0.1:8080: getsockopt: connection refused
```

**Comment by obnoxxx** [comment]
w00t -- now this one works for me!!!

@jarrpa ptal

File: vagrant/roles/common/tasks/main.yml [hunk 1]
```diff
@@ -11,45 +11,61 @@
...
+## TODO: do we need this or want this?
+#- name: update kubernetes
+#  get_url:
+#    url: https://storage.googleapis.com/kubernetes-release/release/v{{ kubernetes_version }}/bin/linux/amd64/{{ item }}
+#    dest: /usr/bin
+#    mode: 0755
+#  with_items:
+#    - kubelet
+#    - kubectl
+#    - kubeadm
```
Comment (jarrpa) [inline vagrant/roles/common/tasks/main.yml]: Yeah, I was always confused by this task. We can just get rid of the comment entirely, it'll be in the git history if we need it again.

File: vagrant/roles/common/tasks/main.yml [hunk 2]
```diff
@@ -11,45 +11,61 @@
...
+
+- name: install fixed kubelet-kubeadm config file
+  copy: src=10-kubeadm.conf dest=/etc/systemd/system/kubelet.service.d/ owner=root group=root mode=644 backup=yes
```
Comment (jarrpa) [inline vagrant/roles/common/tasks/main.yml]: Something changed that requires this, I imagine? What happens without this conf file?

File: vagrant/roles/common/tasks/main.yml (hunk 1 above)
Comment (obnoxxx) [inline vagrant/roles/common/tasks/main.yml]: sure, I can do that...

File: vagrant/roles/common/tasks/main.yml (hunk 2 above)
Comment (obnoxxx) [inline vagrant/roles/common/tasks/main.yml]: you can't run kubectl

Comment (obnoxxx) [inline vagrant/roles/common/tasks/main.yml]: updated the commit msg...

File: vagrant/roles/common/tasks/main.yml (hunk 1 above)
Comment (obnoxxx) [inline vagrant/roles/common/tasks/main.yml]: done

**Comment by obnoxxx** [comment]
updated according to your comments and improved some commit messages

**Comment by jarrpa** [comment]
Okay! I appreciate the commit messages. :) I'll throw this through its paces later today.

**Comment by obnoxxx** [comment]
Now this works again! 
Also gk-deploy runs to completion.

**Comment by jarrpa** [comment]
If you're gonna add the `ansible-step` script, put in some documentation on what it's useful for. Also, be sure that the gk-deploy changes go in their own PR.

**Comment by obnoxxx** [comment]
@jarrpa:
* separate PR for gk-deploy: #236 
* separate PR for ansible-step with comment in the script: #237 
Will update this one..

**Comment by obnoxxx** [comment]
updated.
this one now depends on PR #236

**Comment by jarrpa** [comment]
LGTM :)

**Pull request merged**